manager_test.go

```diff
@@ -263,15 +286,36 @@ var _ = Describe("Extension Manager", func() {
 			eiriniManager.Options.FilterEiriniApps = &filtering
 			fakeCorev1 := &cfakes.FakeCoreV1Interface{}
 			fakePod := &cfakes.FakePodInterface{}
-			opts := &metav1.ListOptions{}
+			fakeWatch := &cfakes.FakeInterface{}
+			var label string
+
 			fakePod.WatchCalls(func(m metav1.ListOptions) (watch.Interface, error) {
-				opts = &m
-				return nil, nil
+				label = m.LabelSelector
+				return fakeWatch, nil
 			})
 			fakeCorev1.PodsCalls(func(s string) corev1client.PodInterface { return fakePod })
-			_, err := eiriniManager.GenWatcher(fakeCorev1)
+
+			w, err := eiriniManager.GenWatcher(fakeCorev1)
+			cw, ok := w.(*watchtools.RetryWatcher)
+			// Give the watcher a chance to get to sending events (blocking)
+			time.Sleep(10 * time.Millisecond)
+
+			cw.Stop()
+
+			select {
+			case <-cw.Done():
+				break
+			case <-time.After(10 * time.Millisecond):
+
+			}
+
+			// RetryWatcher result channel should be closed
+			_, ok = <-cw.ResultChan()
+
+			Expect(ok).ToNot(BeTrue())
+			Expect(label).To(Equal(""))
```

Comment:
Use some default value and check that it was rewritten to avoid false positives (e.g. in case Watch() was never called)